tests/e2e/vmops_test.go

```diff
@@ -0,0 +1,179 @@
+	status = verification.VerifyAttachedStatus(vm.volName1, vm.config.DockerHosts[0], vm.config.EsxHost)
+	c.Assert(status, Equals, true, Commentf("Volume %s is not attached", vm.volName1))
+
+	// 4. Stop container and verify volume is detached
+	_, err = dockercli.RemoveContainer(vm.config.DockerHosts[0], vm.testContainer)
+	c.Assert(err, IsNil, Commentf("Failed to remove container %s with volume %s attached as persistent\n", vm.testContainer, vm.volName1))
+	status = verification.VerifyDetachedStatus(vm.volName1, vm.config.DockerHostNames[0], vm.config.EsxHost)
+	c.Assert(status, Equals, true, Commentf("Volume %s is not attached", vm.volName1))
+
+	misc.LogTestEnd(c.TestName())
+}
+
+// TestVMSnapWithIndependentDisk - Snapshot a VM with a independent
+// disk attached.
+// 1. Run a container with a volume attached as independent
+// 2. Take snapshot of the vm operation should fail as vm has independent disk attached
+// 3. Stop the container
+// 4. Again try taking the snapshot of vm - operation must succeed
+func (vm *VMOpsTest) TestVMSnapWithIndependentDisk(c *C) {
+	misc.LogTestStart(c.TestName())
+	// 1. Start container with volume attached as independent-persistent
+	_, err := dockercli.AttachVolume(vm.config.DockerHosts[0], vm.volName2, vm.testContainer)
+	c.Assert(err, IsNil, Commentf("Failed running container %s with volume %s attached as independent disk\n", vm.testContainer, vm.volName2))
+
+	// 2. Snapshot VM (should fail) and verify attached state of volume
+	_, err = takeSnapshot(vm.config.EsxHost, vm.config.DockerHostNames[0])
+	c.Assert(err, Not(IsNil), Commentf("Expected to fail snapshotting VM %s with volume %s attached as independent disk", vm.config.DockerHostNames[0], vm.volName2))
+
+	status := verification.VerifyAttachedStatus(vm.volName2, vm.config.DockerHosts[0], vm.config.EsxHost)
+	c.Assert(status, Equals, true, Commentf("Volume %s is not attached", vm.volName2))
+
+	// 3. Stop container and verify volume is detached
+	_, err = dockercli.RemoveContainer(vm.config.DockerHosts[0], vm.testContainer)
+	c.Assert(err, IsNil, Commentf("Failed to remove container %s with volume %s attached as independent disk\n", vm.testContainer, vm.volName2))
+	status = verification.VerifyDetachedStatus(vm.volName2, vm.config.DockerHostNames[0], vm.config.EsxHost)
+	c.Assert(status, Equals, true, Commentf("Volume %s is not attached", vm.volName1))
+
+	// 4. Snapshot VM, should succeed
+	_, err = takeSnapshot(vm.config.EsxHost, vm.config.DockerHostNames[0])
+	c.Assert(err, IsNil, Commentf("Failed to snapshot VM %s with no volumes attached", vm.config.DockerHostNames[0]))
+
+	// 5. Remove snapshot
+	_, err = removeSnapshot(vm.config.EsxHost, vm.config.DockerHostNames[0])
+	c.Assert(err, IsNil, Commentf("Failed to remove snapshot of VM %s", vm.config.DockerHostNames[0], vm.volName1))
+	misc.LogTestEnd(c.TestName())
+}
+
+// takeSnapshot - create a snapshot for a VM
+func takeSnapshot(host, name string) (string, error) {
```

Comment:
can you please move out `takeSnapshot` & `removeSnapshot` from test file to util file?